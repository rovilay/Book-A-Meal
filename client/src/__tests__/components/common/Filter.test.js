import React from 'react';
import { shallow } from 'enzyme';
import FilterComp from '../../../components/common/Filter';


describe('Filter component test', () => {
  const setup = () => {
    const props = {
      setFilter: () => jest.fn()
    };

    return shallow(<FilterComp {...props} />)
  };

  it('should render FilterComp correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });

  it('should call `setFilter` on submitting form', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn(),
      target: {
        name: 'date',
        value: '2018-08-27'
      }
    };

    const handleFormSubmitSpy= jest.spyOn(wrapper.instance(), 'handleFormSubmit');
    wrapper.instance().handleFormSubmit(event)
    expect(handleFormSubmitSpy).toHaveBeenCalled();
    done();
  });

  it('should update state on filter input change', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn(),
      target: {
        name: 'filter',
        value: 'month'
      }
    };

    wrapper.find('#filter').simulate('change', event);
    expect(wrapper.instance().state.filter).toEqual('month');

    done();
  });
});
