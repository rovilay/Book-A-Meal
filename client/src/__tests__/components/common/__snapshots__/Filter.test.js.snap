// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component test should render FilterComp correctly 1`] = `
<div
  className="filter"
>
  <form
    onSubmit={[Function]}
  >
    <label
      className="label"
      htmlFor="filter"
    >
      Filter By:
    </label>
    <div
      className="input-div"
    >
      <select
        id="filter"
        name="filter"
        onChange={[Function]}
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="date"
        >
          Date
        </option>
        <option
          value="month"
        >
          Month
        </option>
      </select>
    </div>
    <button
      className="btn-2 filter-btn"
      type="submit"
      value="Search"
    >
       Filter 
    </button>
  </form>
</div>
`;
